### PR TITLE
feat: unbundle delphi-logger and re-export it in delphi-utils

### PR DESCRIPTION
**Pull request by dshemetov**
### Description
Refactor `delphi-utils` to depend on `delphi-logger` standalone repo.

### Changelog

- Moved `logger.py` to the standalone [delphi-logger repo](https://github.com/cmu-delphi/delphi-logger).
- Now it's importable, so import it in `delphi-utils` and re-export it, so that `from delphi-utils import get_structured_logger` still works.

### Fixes 
- Partial fix to https://github.com/cmu-delphi/delphi-epidata/issues/1467#issuecomment-2146373051

### CODE REVIEW

**Comment by dshemetov** [comment]
See this [snag here](https://github.com/cmu-delphi/delphi-epidata/pull/1469#issuecomment-2151038489).

File: _delphi_utils_python/delphi_utils/logger.py [hunk 1]
```diff
@@ -1,255 +1,10 @@
...
-    with multiprocessing.Manager() as manager:
-        logger_thread = LoggerThread(logger, manager.Queue())
-        try:
-            with multiprocessing.Pool(*poolargs) as pool:
-                yield pool, logger_thread.get_sublogger()
-                pool.close()
-                pool.join()
-        finally:
-            logger_thread.stop()
+from delphi_logger import get_structured_logger  # pylint: disable=unused-import
```
Comment (nmdefries) [inline _delphi_utils_python/delphi_utils/logger.py:10]: question: this bit does the reexporting?

**Review by nmdefries — APPROVED** [review]
Couple things to verify, but otherwise looks good.

**Comment by dshemetov** [comment]
Closed for another approach (see comment [here](https://github.com/cmu-delphi/delphi-epidata/issues/1467#issuecomment-2189469399)).

**Pull request closed without merging**